File: VB/Scaffolding.CustomFilter/ViewModels/IssueCollectionViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports Scaffolding.CustomFilter.Common.Utils
Imports Scaffolding.CustomFilter.IssueContextDataModel
Imports Scaffolding.CustomFilter.Common.DataModel
Imports Scaffolding.CustomFilter.Model
Imports Scaffolding.CustomFilter.Common.ViewModel

Namespace Scaffolding.CustomFilter.ViewModels
	Partial Public Class IssueCollectionViewModel
		Inherits CollectionViewModel(Of Issue, Integer, IIssueContextUnitOfWork)
		Public Sub New()
			Me.New(UnitOfWorkSource.GetUnitOfWorkFactory())
		End Sub
		Public Sub New(ByVal unitOfWorkFactory As IUnitOfWorkFactory(Of IIssueContextUnitOfWork))
			MyBase.New(unitOfWorkFactory, Function(x) x.Issues)
		End Sub
	End Class
End Namespace

File: CS/Scaffolding.CustomFilter/ViewModels/IssueCollectionViewModel.cs
using System;
using System.Linq;
using Scaffolding.CustomFilter.Common.Utils;
using Scaffolding.CustomFilter.IssueContextDataModel;
using Scaffolding.CustomFilter.Common.DataModel;
using Scaffolding.CustomFilter.Model;
using Scaffolding.CustomFilter.Common.ViewModel;

namespace Scaffolding.CustomFilter.ViewModels {
    public partial class IssueCollectionViewModel : CollectionViewModel<Issue, int, IIssueContextUnitOfWork> {
        public IssueCollectionViewModel()
            : this(UnitOfWorkSource.GetUnitOfWorkFactory()) {
        }
        public IssueCollectionViewModel(IUnitOfWorkFactory<IIssueContextUnitOfWork> unitOfWorkFactory)
            : base(unitOfWorkFactory, x => x.Issues) {
        }
    }
}